backend/analysis-engine/src/main/java/org/sonarsource/sonarlint/core/analysis/command/AnalyzeCommand.java

```diff
@@ -77,29 +164,57 @@ private AnalysisResults doRunAnalysis(ModuleRegistry moduleRegistry, ProgressMon
         .toList());
     }
     try {
-      var result = moduleContainer.analyze(configuration, issueListener, progressMonitor, trace);
+      var result = moduleContainer.analyze(configuration, issueListener, progressIndicator, trace);
       if (trace != null) {
         trace.setData("failedFilesCount", result.failedAnalysisFiles().size());
         trace.finishSuccessfully();
       }
+      result.setDuration(Duration.ofMillis(System.currentTimeMillis() - startTime));
       return result;
     } catch (Throwable e) {
-      originalException = e;
       if (trace != null) {
         trace.finishExceptionally(e);
       }
       throw e;
-    } finally {
-      try {
-        if (moduleContainer.isTransient()) {
-          moduleContainer.stopComponents();
-        }
-      } catch (Exception e) {
-        if (originalException != null) {
-          e.addSuppressed(originalException);
-        }
-        throw e;
+    }
+  }
+
+  public AnalyzeCommand mergeWith(AnalyzeCommand otherNewerAnalyzeCommand) {
+    var analysisConfiguration = configurationSupplier.get();
+    var newerAnalysisConfiguration = otherNewerAnalyzeCommand.configurationSupplier.get();
+    var mergedInputFiles = new ArrayList<>(newerAnalysisConfiguration.inputFiles());
+    var newInputFileUris = newerAnalysisConfiguration.inputFiles().stream().map(ClientInputFile::uri).collect(Collectors.toSet());
+    for (ClientInputFile inputFile : analysisConfiguration.inputFiles()) {
+      if (!newInputFileUris.contains(inputFile.uri())) {
```

Comment:
Looks like we missing test for this case. Code appears to be correct though.

Reply:
Yes, this part is not covered, apparently I was lazy when implementing it... :disappointed:
Batching is not working properly though, that's what I discovered when testing with the LS, so even the code probably has a problem... I'm looking into it better coverage and fixing the bug now, and will open a separate PR